plugins/modules/vmware_dvswitch_info.py

```diff
@@ -0,0 +1,313 @@
+      [
+        {
+          "configure": {
+            "folder": "network",
+            "hosts": [
+              "esxi-test-02.local",
+              "esxi-test-01.local"
+            ],
+            "settings": {
+              "healthCheck": {
+                "TeamingHealthCheckConfig": false,
+                "VlanMtuHealthCheckConfig": false
+              },
+              "netflow": {
+                "activeFlowTimeout": 60,
+                "collectorIpAddress": "",
+                "collectorPort": 0,
+                "idleFlowTimeout": 15,
+                "internalFlowsOnly": false,
+                "observationDomainId": 0,
+                "samplingRate": 0,
+                "switchIpAddress": null
+              },
+              "properties": {
+                "administratorContact": {
+                  "contact": null,
+                  "name": null
+                },
+                "advanced": {
+                  "maxMtu": 1500,
+                  "multicastFilteringMode": "legacyFiltering"
+                },
+                "discoveryProtocol": {
+                  "operation": "listen",
+                  "protocol": "cdp"
+                },
+                "general": {
+                  "ioControl": true,
+                  "name": "DVSwitch01",
+                  "numPorts": 10,
+                  "numUplinks": 1,
+                  "vendor": "VMware, Inc.",
+                  "version": "6.6.0"
+                }
+              },
+              "privateVlan": []
+            }
+          }
+        }
+      ]
+'''
+
+try:
+    from pyVmomi import vim
+except ImportError:
+    pass
+
+from ansible_collections.community.vmware.plugins.module_utils.vmware import PyVmomi, vmware_argument_spec, find_obj, find_object_by_name
+from ansible.module_utils.basic import AnsibleModule
+
+
+class VMwareDvSwitchInfoManager(PyVmomi):
+    def __init__(self, module):
+        super(VMwareDvSwitchInfoManager, self).__init__(module)
+        self.folder = self.params['folder']
+        self.switch_name = self.params['switch_name']
+
+        folder_obj = None
+        if self.folder:
+            folder_obj = self.content.searchIndex.FindByInventoryPath(self.folder)
+            if not folder_obj:
+                self.module.fail_json(msg="Failed to find folder specified by %s" % self.folder)
+
+        if self.switch_name:
+            self.switch_objs = [find_object_by_name(self.content, self.switch_name, vim.DistributedVirtualSwitch, folder_obj)]
+            if None in self.switch_objs:
+                self.switch_objs = None
+        else:
+            self.switch_objs = find_obj(self.content, [vim.DistributedVirtualSwitch], '', first=False)
+
+        if not self.switch_objs:
+            if self.switch_name:
+                self.module.fail_json(msg="Distributed Virtual Switch %s not found" % self.switch_name)
+            else:
+                self.module.fail_json(msg="Distributed Virtual Switch not found")
```

Comment:
I think we should return empty list rather than failing.

Reply:
okay :)
I was worrying whether to make the return value an error or an empty list, I'll make the return value an empty list.